### PR TITLE
Refactor server setup & management utilities.

### LINKED ISSUE #6072: Ensure Addressables are written to an application data folder on all platforms B: 80
### Environment
- Game Version:  21021823
- OS Version:  Ubuntu Server 20.04 LTS

### Detailed description of bug
If AddressableCatalogues or LobbyAddressableCatalogues are defined to download in config.json then server will download all addressables in directory named com.unity.addressables which located in / (root directory of the OS). It is a security vulnerability.
All addressables should be located under gaming server files.

### Steps to reproduce
1. Define AddressableCatalogues  or LobbyAddressableCatalogues in config.json
2. Start up a server
3. ls -a /

### PR DESCRIPTION
**Pull request by craftxbox**
### Purpose
This PR pulls the installation and management helper scripts onto the repository, as previously they were only available through the CDN.
The scripts have also been majorly refactored as they have not been updated since April of 2020!
This introduces a new utility, manageuss, intended to *SECURELY* control the status of the systemd unitystation unit as an unprivileged user.

Fixes #6072 as it was caused entirely by supervisord, which has been thrown out in favor of systemd.

## **__/!\\__** ***Do NOT overlook this*** **__/!\\__**
In order for manageuss to work properly, your unit MUST be named `unitystation.service`, and, the binary MUST be chmod 4711 root:root
Any other configuration will either cause the program to fail or worse, introduce a serious privilege escalation vulnerability in your server. 

### Changelog:
Does not change game code.

### CODE REVIEW

**Comment by craftxbox** [comment]
TODO: 
- [x] manageuss
- [x] unitystation.service
- [x] installserver.sh
- [x] update.sh
- [x] editconfig.sh

File: Tools/server-utils/update.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        echo "PLEASE DEESCALATE TO NON-ADMINISTRATOR BEFORE CONTINUING!!!"
+        echo "IF THIS SERVER NORMALLY RUNS AS ROOT, YOUR SETUP IS UNSUPPORTED AND CANNOT BE UPDATED USING THIS TOOL!"
+        exit 1;
+fi
+
+SDIR='us13' # Change this if your server directory differs!!!!
+SVER=$(wget -qO- https://api.unitystation.org/latestbuild)
+SAST='Unitystation_Data/StreamingAssets'
+BINFO=$(<"./$SDIR/$SAST/buildinfo.json")
+LVER=`echo $BINFO | jq '.BuildNumber'`
```
Comment (craftxbox) [inline Tools/server-utils/update.sh]: Not quite working, but, pretty close. Will fix this more in the morning.

**Comment by corp-0** [comment]
thank you for adding documentation for the new tool. I do think though it needs more information and less assumption on what the users know, try giving them a step by step including one step where you copy over the systemd unit which you also included and how they should execute the tool.

**Comment by craftxbox** [comment]
> thank you for adding documentation for the new tool. I do think though it needs more information and less assumption on what the users know, try giving them a step by step including one step where you copy over the systemd unit which you also included and how they should execute the tool.

Sure! The setup script is pretty much completely self contained, It only needs to be run and it *should* take care of everything (ex, the systemd stuff is entirely transparent to the user). I'll add a bit of extra basic steps though.

**Review by corp-0 — DISMISSED** [review]

**Review by corp-0 — APPROVED** [review]

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**